File: backend/src/compliance_platform/api/identity.py
"""Who the server thinks you are.

Exists because attribution stopped being something the client supplies
(ADR-0061). The evidence-request forms used to ask "your name" and send
it; the server now ignores that in favour of the identity the proxy
authenticated. Removing the field without replacing it would leave a
reviewer recording decisions under a name they cannot see -- so the UI
asks the server instead, and shows the answer.

Also the fastest way for an operator to confirm the proxy is actually
forwarding the header: if this returns `unauthenticated` through the
deployed stack, `X-Remote-User` is not arriving and every decision is
being recorded anonymously.
"""

from __future__ import annotations

from fastapi import APIRouter, Depends
from pydantic import BaseModel

from compliance_platform.core.identity import UNAUTHENTICATED_ACTOR, get_actor

router = APIRouter(tags=["identity"])


class Identity(BaseModel):
    actor: str
    # Explicit rather than making the client compare against a magic
    # string it would have to keep in sync with the backend.
    is_authenticated: bool


@router.get("/identity", response_model=Identity)
def read_identity(actor: str = Depends(get_actor)) -> Identity:
    return Identity(actor=actor, is_authenticated=actor != UNAUTHENTICATED_ACTOR)

File: frontend/src/api/identity.ts
import { useQuery } from '@tanstack/react-query'
import { apiClient } from './client'
import type { Identity } from './types'

/**
 * Who the server will attribute this session's decisions to (ADR-0061).
 *
 * The evidence-request forms used to ask "your name" and send it. The
 * server now ignores that in favour of the identity the reverse proxy
 * authenticated, so the field was removed — and removing it without
 * replacing it would leave a reviewer recording decisions under a name
 * they cannot see. This is the replacement: ask the server, and show
 * the answer where the input used to be.
 *
 * Cached for the session. It cannot change without the browser
 * re-authenticating, which reloads the page anyway.
 */
export function useIdentity() {
  return useQuery({
    queryKey: ['identity'],
    queryFn: () => apiClient.get<Identity>('/identity'),
    staleTime: Infinity,
  })
}
